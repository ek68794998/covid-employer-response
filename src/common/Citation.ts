import { CitationSource } from "./CitationSource";
import { CitationType } from "./CitationType";

export class Citation {
	public positivity: number = 0;

	public sources?: CitationSource[];

	public summary: string = "";

	public type?: CitationType;
}
