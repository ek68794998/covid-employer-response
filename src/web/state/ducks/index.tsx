import { combineReducers } from "redux";

import { getLocalizedStrings } from "./localization/reducers";

export default combineReducers({
	strings: getLocalizedStrings,
});