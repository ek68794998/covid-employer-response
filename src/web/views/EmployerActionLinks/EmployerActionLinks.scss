a.EmployerActionLinks__Link {
	line-height: 0;
	margin-right: 8px;
	opacity: 0;
	transition: 0.2s opacity;
}

@media (any-hover: none) {
	a.EmployerActionLinks__Link {
		opacity: 1.0 !important;
	}
}
