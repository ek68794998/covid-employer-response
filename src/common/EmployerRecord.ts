import { Citation } from "./Citation";
import { EmployerLocation } from "./EmployerLocation";
import { EmployerRating } from "./EmployerRating";

export class EmployerRecord {
	public citations: Citation[] = [];

	public employeesBeforeMax: number = -1;

	public employeesBeforeMin: number = -1;

	public id: string = "";

	public image?: string;

	public location: EmployerLocation = new EmployerLocation();

	public name: string = "";

	public officialWebsite?: string;

	public rating: EmployerRating = "fair";

	public summary: string = "";

	public wiki?: string;
}
