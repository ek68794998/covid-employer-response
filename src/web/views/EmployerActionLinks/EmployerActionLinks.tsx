import React from "react";
import { useSelector } from "react-redux";
import { RouteProps } from "react-router-dom";

import { ProjectUrl } from "../../../common/constants/UrlConstants";
import { DesignHelpers } from "../../../common/DesignHelpers";
import { EmployerRecordMetadata } from "../../../common/EmployerRecordMetadata";
import { LocalizedStrings } from "../../../common/LocalizedStrings";
import { WikipediaHelpers } from "../../../common/WikipediaHelpers";

import { getStrings } from "../../state/ducks/localization/selectors";

import "./EmployerActionLinks.scss";

interface Props extends RouteProps {
	employer: EmployerRecordMetadata;

	hideEdit?: boolean;

	hideProfileLink?: boolean;

	hideWebsite?: boolean;

	hideWikipedia?: boolean;
}

const getEmployerEditComponent =
	(employer: EmployerRecordMetadata, strings: LocalizedStrings): JSX.Element | null => {
		const editUrl: string = `${ProjectUrl}/blob/master/public/employers/${employer.id}.yml`;

		return (
			<a
				className="EmployerActionLinks__Link"
				href={editUrl}
				rel="noopener noreferrer"
				target="_blank"
				title={strings.detailDescriptions.edit}
			>
				{DesignHelpers.materialIcon("edit")}
			</a>
		);
	};

const getEmployerProfileLinkComponent =
	(employer: EmployerRecordMetadata, strings: LocalizedStrings): JSX.Element | null => {
		if (!employer.id) {
			return null;
		}

		const link: string = `/employers/${employer.id}`;

		return (
			<a
				className="EmployerActionLinks__Link"
				href={link}
				title={strings.detailDescriptions.linkToEmployer}
			>
				{DesignHelpers.materialIcon("link")}
			</a>
		);
	};

const getEmployerWebsiteComponent =
	(employer: EmployerRecordMetadata, strings: LocalizedStrings): JSX.Element | null => {
		if (!employer.officialWebsite) {
			return null;
		}

		return (
			<a
				className="EmployerActionLinks__Link"
				href={employer.officialWebsite}
				rel="noopener noreferrer"
				target="_blank"
				title={strings.detailDescriptions.officialWebsite}
			>
				{DesignHelpers.materialIcon("home")}
			</a>
		);
	};

const getEmployerWikipediaComponent =
	(employer: EmployerRecordMetadata, strings: LocalizedStrings): JSX.Element | null => {
		const employerWikipediaUrl: string | null = WikipediaHelpers.getWikipediaUrl(employer.wiki);

		if (!employerWikipediaUrl) {
			return null;
		}

		return (
			<a
				className="EmployerActionLinks__Link"
				href={employerWikipediaUrl}
				rel="noopener noreferrer"
				target="_blank"
				title={strings.detailDescriptions.wikipedia}
			>
				{DesignHelpers.materialIcon("language")}
			</a>
		);
	};

const EmployerActionLinks: React.FC<Props> = (props: Props): React.ReactElement => {
	const strings: LocalizedStrings = useSelector(getStrings);
	const { employer, hideEdit, hideProfileLink, hideWebsite, hideWikipedia } = props;

	return (
		<>
			{!hideWikipedia && getEmployerWikipediaComponent(employer, strings)}
			{!hideWebsite && getEmployerWebsiteComponent(employer, strings)}
			{!hideEdit && getEmployerEditComponent(employer, strings)}
			{!hideProfileLink && getEmployerProfileLinkComponent(employer, strings)}
		</>
	);
};

export default EmployerActionLinks;
