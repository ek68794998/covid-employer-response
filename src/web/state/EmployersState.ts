import { EmployerRecord } from "../../common/EmployerRecord";

export interface EmployersState {
	items: EmployerRecord[];
}