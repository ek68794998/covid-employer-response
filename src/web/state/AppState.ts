import { EmployersState } from "./EmployersState";

import { LocalizedStrings } from "../../common/LocalizedStrings";

export interface AppState {
	employers: EmployersState;

	environment: string;

	strings: LocalizedStrings;
}