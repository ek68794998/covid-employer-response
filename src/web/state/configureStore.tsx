import { AnyAction, applyMiddleware, compose, createStore, Store } from "redux";
import thunk from "redux-thunk";

import { Production } from "../../common/constants/EnvironmentConstants";

import { AppState } from "./AppState";
import rootReducer from "./ducks";

const configureStore = (preloadedState: Partial<AppState>): Store<AppState, AnyAction> => {
	const isProd: boolean = process.env.NODE_ENV === Production;
	const composeEnhancers: any =
		(!isProd && typeof window !== "undefined" && (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;

	const store: Store<AppState, AnyAction> = createStore(
		rootReducer,
		preloadedState,
		composeEnhancers(applyMiddleware(thunk)),
	);

	if (module.hot) {
		// Enable Webpack hot module replacement for reducers
		module.hot.accept("./ducks", () => {
			const nextRootReducer: any = require("./ducks").default;
			store.replaceReducer(nextRootReducer);
		});
	}

	return store;
};

export default configureStore;