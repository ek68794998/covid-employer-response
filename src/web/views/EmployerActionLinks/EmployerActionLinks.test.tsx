import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import renderer, { ReactTestRendererJSON } from "react-test-renderer";
import { AnyAction, Store } from "redux";

import { EmployerRecord } from "../../../common/EmployerRecord";

import { AppState } from "../../state/AppState";
import configureStore from "../../state/configureStore";

import EmployerActionLinks from "./EmployerActionLinks";

describe("<EmployerActionLinks />", () => {
	test("renders without exploding", () => {
		const store: Store<AppState, AnyAction> = configureStore({
			strings: {
				detailDescriptions: {
					edit: "edit string",
					employees: "employees string",
					location: "location string",
					officialWebsite: "officialWebsite string",
					rating: "rating string",
					wikipedia: "wikipedia string",
				},
			},
		});

		const renderedValue: ReactTestRendererJSON | null =
			renderer.create(
				<Provider store={store}>
					<BrowserRouter>
						<EmployerActionLinks employer={new EmployerRecord()} />
					</BrowserRouter>
				</Provider>,
			).toJSON();

		expect(renderedValue).toMatchSnapshot();
	});
});
