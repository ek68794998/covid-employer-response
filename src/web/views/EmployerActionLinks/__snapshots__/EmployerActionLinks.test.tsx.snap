// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmployerActionLinks /> renders without exploding 1`] = `null`;
